### PR TITLE
Freertos upd cfg

### PR DESCRIPTION
**Pull request by phmoussay**
-Changes on lds generator for FreeRTOS
-Changes on FreeRTOS (pmp, config header generator)
-New FreeRTOS example with pmp
-Warning and bug fix on FreeRTOS port

### CODE REVIEW

File: scripts/libmetal.mk [hunk 1]
```diff
@@ -80,6 +84,7 @@ metal-bsp:\
 	   $(BSP_DIR)/metal.default.lds \
 	   $(BSP_DIR)/metal.ramrodata.lds \
 	   $(BSP_DIR)/metal.scratchpad.lds \
+	   $(BSP_DIR)/metal.privilege.lds \
```
Comment (bsousi5) [inline scripts/libmetal.mk]: where is privilege.lds come from? you meant freertos?

Comment (phmoussay) [reply scripts/libmetal.mk]: Yes, that's it, i miss this one.

**Comment by phmoussay** [comment]
Just in case, I passed product core IP test:
===== TEST RESULTS =====
Pass "e20: succeeded"
Pass "e21: succeeded"
Pass "e24: succeeded"
Pass "e31: succeeded"
Pass "e34: succeeded"
Pass "e76: succeeded"
Pass "e76mc: succeeded"
Pass "s21: succeeded"
Pass "s51: succeeded"
Pass "s54: succeeded"
Pass "s76: succeeded"
Pass "s76mc: succeeded"
Pass "u52: succeeded"
Pass "u54: succeeded"
Pass "u54mc: succeeded"
Pass "u74: succeeded"
Pass "u74mc: succeeded"
Pass "u84: succeeded"

File: scripts/libmetal.mk [hunk 2]
```diff
@@ -43,6 +43,10 @@ $(BSP_DIR)/metal.scratchpad.lds: $(BSP_DIR)/design.dts $(LDSCRIPT_GENERATOR)
 	$(MAKE) -f scripts/virtualenv.mk virtualenv
 	. $(FREEDOM_E_SDK_VENV_PATH)/bin/activate && $(LDSCRIPT_GENERATOR) -d $< -o $@ --scratchpad
 
+$(BSP_DIR)/metal.freertos.lds: $(BSP_DIR)/design.dts $(LDSCRIPT_GENERATOR)
+	$(MAKE) -f scripts/virtualenv.mk virtualenv
+	. venv/bin/activate && $(LDSCRIPT_GENERATOR) -d $< -o $@ --freertos
```
Comment (nategraff-sifive) [inline scripts/libmetal.mk]: should be `. $(FREEDOM_E_SDK_VENV_PATH)/bin/activate`

Comment (phmoussay) [reply scripts/libmetal.mk]: Done

**Comment by phmoussay** [comment]
Ok, merge conflicts solved, I just added one last commit to bump a minor change on example-freertos-pmp-blinky.

**Pull request merged**